src/app/app-platform/ContentAppPlatform.cpp

```diff
@@ -100,37 +100,67 @@ EndpointId ContentAppPlatform::AddContentApp(ContentApp * app, EmberAfEndpointTy
         if (mContentApps[index] == app)
         {
             ChipLogProgress(DeviceLayer, "Already added");
+            // already added, return endpointId of already added endpoint.
+            // desired endpointId does not have any impact
             return app->GetEndpointId();
         }
         index++;
     }
 
+    if (desiredEndpointId >= emberAfFixedEndpointCount() &&
+        emberAfGetDynamicIndexFromEndpoint(desiredEndpointId) != kEmberInvalidEndpointIndex)
+    {
+        // an endpoint already exists with the desiredEndpointId. return error.
+        ChipLogError(DeviceLayer, "Failed to add dynamic endpoint: desired endpointID already in use!");
+        return kNoCurrentEndpointId;
+    }
+
     index = 0;
     while (index < CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT)
     {
         if (nullptr == mContentApps[index])
```

Comment:
This is a very large indent here. Should we instead have:

```cpp

if (mContentApps[index] != nullptr) {
  index++;
  continue;
}
```

Reply:
done.